src/daft-scan/src/python.rs

```diff
@@ -1,4 +1,44 @@
-use pyo3::prelude::*;
+use pyo3::{prelude::*, types::PyTuple};
+use serde::{Deserialize, Serialize};
+
+use crate::py_object_serde::{deserialize_py_object, serialize_py_object};
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+struct PyObjectSerializableWrapper(
+    #[serde(
+        serialize_with = "serialize_py_object",
+        deserialize_with = "deserialize_py_object"
+    )]
+    pub PyObject,
+);
+
+/// Python arguments to a Python function that produces Tables
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct PythonTablesFactoryArgs(Vec<PyObjectSerializableWrapper>);
+
+impl PythonTablesFactoryArgs {
+    pub fn new(args: Vec<PyObject>) -> Self {
+        Self(args.into_iter().map(PyObjectSerializableWrapper).collect())
+    }
+
+    pub fn to_pytuple<'a>(&self, py: Python<'a>) -> &'a PyTuple {
+        pyo3::types::PyTuple::new(py, self.0.iter().map(|x| x.0.as_ref(py)))
+    }
+}
+
+impl PartialEq for PythonTablesFactoryArgs {
+    fn eq(&self, other: &Self) -> bool {
+        if self.0.len() != other.0.len() {
+            return false;
+        }
+        Python::with_gil(|py| {
+            self.0
```

Comment:
Why do we need to implement PartialEq here? That may be very expensive.

Reply:
Synced offline: this is requried for some reason on our PhysicalPlan